_ml-commons-plugin/tutorials/generate-embeddings.md

```diff
@@ -0,0 +1,337 @@
+---
+layout: default
+title: Generating embeddings
+parent: Tutorials
+nav_order: 55
+---
+
+# Generating embeddings for arrays of objects
+
+This tutorial illustrates generating embeddings for arrays of objects.
+
+Replace the placeholders starting with the prefix `your_` with your own values.
+{: .note}
+
+## Step 1: Register an embedding model
+
+For this tutorial, you will use the [Amazon Bedrock Titan Embedding model](https://docs.aws.amazon.com/bedrock/latest/userguide/titan-embedding-models.html). 
+
+
+First, follow the [Amazon Bedrock Titan blueprint example](https://github.com/opensearch-project/ml-commons/blob/2.x/docs/remote_inference_blueprints/bedrock_connector_titan_embedding_blueprint.md) to register and deploy the model. 
+
+Test the model, providing the model ID:
+
+```json
+POST /_plugins/_ml/models/your_embedding_model_id/_predict
+{
+    "parameters": {
+        "inputText": "hello world"
+    }
+}
+```
+{% include copy-curl.html %}
+
+The response contains inference results:
+
+```json
+{
+  "inference_results": [
+    {
+      "output": [
+        {
+          "name": "sentence_embedding",
+          "data_type": "FLOAT32",
+          "shape": [ 1536 ],
+          "data": [0.7265625, -0.0703125, 0.34765625, ...]
+        }
+      ],
+      "status_code": 200
+    }
+  ]
+}
+```
+
+## Step 2: Create an ingest pipeline
+
+The next set of steps will create an ingest pipeline for generating embeddings.
+
+### Step 2.1: Create a k-NN index
+
+First, create a k-NN index:
+
+```json
+PUT my_books
+{
+  "settings" : {
+      "index.knn" : "true",
+      "default_pipeline": "bedrock_embedding_foreach_pipeline"
+  },
+  "mappings": {
+    "properties": {
+      "books": {
+        "type": "nested",
+        "properties": {
+          "title_embedding": {
+            "type": "knn_vector",
+            "dimension": 1536
+          },
+          "title": {
+            "type": "text"
+          },
+          "description": {
+            "type": "text"
+          }
+        }
+      }
+    }
+  }
+}
+```
+{% include copy-curl.html %}
+
+### Step 2.2: Create an ingest pipeline
```

Comment:
Should "ingest" be "inner"?

Reply:
I don't think the heading should be changed but I clarified the intro sentence.